### DISCUSSION TITLE
Conditional on Taxonomy Term not working

### DISCUSSION THREAD

**Discussion by ssyberg** [statamic/cms#6488] [Troubleshooting] [answered]
I tried following the instructions in this [discussion](https://statamic.com/forum/5404-clearing-taxonomies) but I'm still having no luck, I've tried every combo I can think of and nothing seems to be working, is this a new bug perhaps?

I've tried:
```
          if:
            role: 'contains intern'
```

and

```
          if_any:
            role: 'not role::intern'
```

and

```
          if:
            role: 'contains role::intern'
```

And here's the actual entry I'm editing
```
---
id: 52b8f050-30b8-4bd9-90d4-1a6e68ca4a7d
blueprint: people
name: 'Mr Guy'
title: 'Cool Dude'
role: intern
updated_by: 2102ddd8-ec83-43c7-aca4-688507064962
updated_at: 1660331809
published: true
---
```

**Comment by edalzell** [comment]
What is the taxonomy handle?

**Reply by ssyberg** [reply]
We eventually figured this out, we were referencing the taxonomy incorrectly, would be great if this were documented! Also worth mentioning that `contains` works but `not` does not.

**Reply by edalzell** [reply]
Please make an issue or even better a PR to fix!

https://github.com/statamic/docs

**Reply by ssyberg** [reply]
> Please make an issue or even better a PR to fix!
> 
> https://github.com/statamic/docs

I would but I don't feel like I have the expertise here to fully document this, it's still a little hazy to me (especially given that I couldn't figure out the `not` solution). Someone who has a better grasp would be better equipped to document this.

**Comment by jesseleite** [comment]
Because your data value is a string instead of an array:

```
role: intern
```

You should be able:

```
if:
  role: 'equals intern'
```

Or:

```
if:
  role: intern
```

Both are equivalent, and should work in your case.

**Comment by ssyberg** [comment]
Just popping in to note that this happened to me again and I don't remember how I solved it! 😆

**Reply by ssyberg** [reply, marked as answer]
For the next time I encounter this documentation bug, I think this is the answer, though I'm not totally sure if these are the right identifiers. The secret sauce here missing from the docs is the `<some sort of handle>::`

```
if:
  <taxonomny_terms_field_handle>: 'contains <taxonomy_handle???>::<taxonmy_term_slug>'
```

So in my current example it looks like this:
```
unless:
  news_types: 'contains news_types::media-mentions'
```

**Reply by edalzell** [reply]
any chance you could PR that into the docs @ssyberg?

**Reply by ssyberg** [reply]
@edalzell as I mentioned in the thread above, I don't feel like I have a good enough handle on this to accurately document what's going on, I think it would make more sense for someone who better understands this to add. I have no idea what the `::` syntax even is or how/why this works (especially given that `::` does not exist in the very content that is used to evaluate the conditional).

**Reply by duncanmcclean** [reply]
Even if you're just able to open an issue on the [`docs`](https://github.com/statamic/docs/issues) issue, that would be a great help!

**Reply by ssyberg** [reply]
Didn't even know that existed, done! https://github.com/statamic/docs/issues/1450

**Reply by duncanmcclean** [reply]
Thanks!
